### PR TITLE
[FIX] Cancel lines in old revisions

### PR DESCRIPTION
**Pull request by JuaniFreedoo**
[8.0] Cancel lines in old revision orders

### CODE REVIEW

File: sale_order_revision/model/sale_order.py [hunk 1]
```diff
@@ -90,7 +90,10 @@ def copy(self, default=None):
                 'current_revision_id': self.id,
                 'unrevisioned_name': self.unrevisioned_name,
             })
-        return super(sale_order, self).copy(default=default)
+            prev_order = super(sale_order, self).copy(default=default)
```
Comment (pedrobaeza) [inline sale_order_revision/model/sale_order.py]: You always have to call super, not only in the `if` block.

File: sale_order_revision/model/sale_order.py [hunk 2]
```diff
@@ -90,7 +90,10 @@ def copy(self, default=None):
...
+            for line in prev_order.order_line:
+                line.write({'state': 'cancel'})
```
Comment (pedrobaeza) [inline sale_order_revision/model/sale_order.py]: Do directly `prev_order.order_line.write({'state': 'cancel'})`

File: sale_order_revision/model/sale_order.py (hunk 1 above)
Comment (pedrobaeza) [reply sale_order_revision/model/sale_order.py]: This is still pending. Please change also module version number.

File: sale_order_revision/model/sale_order.py [hunk 3]
```diff
@@ -90,7 +90,9 @@ def copy(self, default=None):
                 'current_revision_id': self.id,
                 'unrevisioned_name': self.unrevisioned_name,
             })
-        return super(sale_order, self).copy(default=default)
+        prev_order = super(sale_order, self).copy(default=default)
+        prev_order.order_line.write({'state': 'cancel'})
```
Comment (pedrobaeza) [inline sale_order_revision/model/sale_order.py]: This should be executed only if self.env.context.get('new_sale_revision'):
```
if self.env.context.get('new_sale_revision'):
    prev_order.order_line.write({'state': 'cancel'})
```

**Comment by rousseldenis** [comment]
@JuaniFreedoo Is this still relevant? This will be closed as no activity

**Comment by rousseldenis** [comment]
Closing as very old. Please feel free to reopen if relevant.

**Pull request closed without merging**